integration-tests/src/test_loop/builder.rs

```diff
@@ -331,3 +372,73 @@ impl TestLoopBuilder {
         }
     }
 }
+
+/// Handler to drop all network messages relevant to chunk validated by
+/// `validator_of_chunks_to_drop`. If number of nodes on chain is significant
+/// enough (at least three?), this is enough to prevent chunk from being
+/// included.
+///
+/// This logic can be easily extended to dropping chunk based on any rule.
+pub fn partial_encoded_chunks_dropper(
+    chunks_storage: Arc<Mutex<TestLoopChunksStorage>>,
+    epoch_manager_adapter: Arc<dyn EpochManagerAdapter>,
+    validator_of_chunks_to_drop: AccountId,
+) -> Arc<dyn Fn(NetworkRequests) -> Option<NetworkRequests>> {
+    Arc::new(move |request| {
+        // Filter out only messages related to distributing chunk in the
+        // network; extract `chunk_hash` from the message.
+        let chunk_hash = match &request {
+            NetworkRequests::PartialEncodedChunkRequest { request, .. } => {
+                Some(request.chunk_hash.clone())
+            }
+            NetworkRequests::PartialEncodedChunkResponse { response, .. } => {
+                Some(response.chunk_hash.clone())
+            }
+            NetworkRequests::PartialEncodedChunkMessage { partial_encoded_chunk, .. } => {
+                Some(partial_encoded_chunk.header.chunk_hash())
+            }
+            NetworkRequests::PartialEncodedChunkForward { forward, .. } => {
+                Some(forward.chunk_hash.clone())
+            }
+            _ => None,
+        };
+
+        let Some(chunk_hash) = chunk_hash else {
+            return Some(request);
+        };
+
+        let (chunk, min_chunk_height) = {
+            let chunks_storage = chunks_storage.lock().unwrap();
+            (chunks_storage.get(&chunk_hash).unwrap().clone(), chunks_storage.min_chunk_height)
+        };
+        let prev_block_hash = chunk.prev_block_hash();
+        let shard_id = chunk.shard_id();
+        let height_created = chunk.height_created();
+
+        // If we don't have block on top of which chunk is built, we can't
+        // retrieve epoch id.
+        // This case appears to be too rare to interfere with the goal of
+        // dropping chunk.
+        let Ok(epoch_id) = epoch_manager_adapter.get_epoch_id_from_prev_block(prev_block_hash)
+        else {
+            return Some(request);
+        };
+
+        // If chunk height is too low, don't drop chunk, allow the chain to
+        // warm up.
+        if min_chunk_height.is_some_and(|min_height| height_created <= min_height + 3) {
```

Comment:
the calling of this function is currently hidden in the setup methods (setup_network) and controlled by the params in the test loop builder. Would it be possible to instead calling a method like start_dropping_chunks(...conditions...) in the test body (and pair with stop_dropping_chunks) to control when exactly the behavior will happen?

Reply:
Hm. I tried another option - what if `TestLoopChunksStorage` will control it instead? This makes condition content well hidden, and it is needed only to allow chain warmup anyway.